### PR TITLE
Update iiiftilesource.js

### PR DESCRIPTION
**Pull request by filak**
Fix for https://github.com/openseadragon/openseadragon/issues/2198

### CODE REVIEW

File: src/iiiftilesource.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ $.IIIFTileSource = function( options ){
 
     if (!options.maxLevel && !this.emulateLegacyImagePyramid) {
         if (!this.scale_factors) {
-            options.maxLevel = Number(Math.ceil(Math.log(Math.max(this.width, this.height), 2)));
+            options.maxLevel = Number(Math.round(Math.log(Math.max(this.width, this.height), 2)));
```
Comment (iangilman) [inline src/iiiftilesource.js:137]: I suspect this one should stay `ceil`. This is determining how many levels we have, so it makes sense to round up. The other ones all have to do with pixel width/height, so they are a different thing.

**Review by iangilman — COMMENTED** [review]
@filak Thank you for making this PR! Have you been able to test it? Did it work to fix your issue? 

@azaroth42 Can you take a look? More things have been changed than you suggested, but they seem like they might all be worth changing (except for the one I've called out). What do you think?

File: src/iiiftilesource.js (hunk 1 above)
Comment (filak) [reply src/iiiftilesource.js:137]: I tested the updated code and it works as expected - no drop in the last tile column.

I ran the test suite and there were no errors concerning the iiiftilesource.js

I not completely sure about the line you have mentioned either but I have not seen any misbehaviour.

Comment (filak) [reply src/iiiftilesource.js:137]: The other option uses round as well: https://github.com/openseadragon/openseadragon/blob/ce45e34265d27de835ae40e6b486deb14f70c6ba/src/iiiftilesource.js#L140

So IMHO it kinda makes sense to unify the approach...

Comment (iangilman) [reply src/iiiftilesource.js:137]: Oh! Interesting... I didn't realize the other path was using round already. @azaroth42 What do you think?

Comment (azaroth42) [reply src/iiiftilesource.js:137]: Apologies, work-work is snowing me at the moment. Perhaps @tomcrane ?

Comment (iangilman) [reply src/iiiftilesource.js:137]: @azaroth42 Thanks for the lead! If I don't hear from @tomcrane in a while, I'll ping you again.

**Comment by iangilman** [comment]
@azaroth42 @tomcrane Can either of you take a look at this PR to make sure it's sensible for improved IIIF support? If not, is there anyone else in the IIIF team who might be able to?

**Review by iangilman — APPROVED** [review]
Okay, I've talked with the folks in the IIIF Slack, and no one's quite sure if that one `round` is the right thing, but I figure we might as well merge it and see how it goes! 

@filak Apologies for the delay on this. Thank you again for tracking it down and making the fix!

**Comment by iangilman** [comment]
@stephenwf Good question! Is that something you can test, or tell me how to test? Which line are you thinking of (a lot of lines have been changed)?

**Pull request merged**